### PR TITLE
Feature/2950  add popup at login if username null

### PR DESCRIPTION
**Pull request by gbonaspetti**
#2950 

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: frontend/src/container/Home.jsx [hunk 1]
```diff
@@ -86,6 +207,77 @@ export class Home extends React.Component {
...
+                    </>
+                  )}
+
+                  <div className='homepage__usernamePopup__body__checkbox'>
+                    <input
+                      className='homepage__usernamePopup__body__checkbox__input'
+                      type='checkbox'
+                      onChange={this.handleClickCheckbox}
+                    />
+                    {props.t('never ask me again')}
```
Comment (raphj) [inline frontend/src/container/Home.jsx:264]: Should this be capitalized?

File: frontend/src/container/Home.jsx [hunk 2]
```diff
@@ -86,6 +207,77 @@ export class Home extends React.Component {
...
+                  </div>
+                  <div className='homepage__usernamePopup__body__smallmsg'>
+                    ({props.t('you can always set your username in your account preferences')})
```
Comment (raphj) [inline frontend/src/container/Home.jsx:267]: Should this be capitalized?

File: frontend/src/helper.js [hunk 3]
```diff
@@ -86,7 +89,8 @@ export const CONTENT_TYPE = {
 export const COOKIE_FRONTEND = {
   LAST_CONNECTION: 'lastConnection',
   DEFAULT_LANGUAGE: 'defaultLanguage',
-  DEFAULT_EXPIRE_TIME: 180
+  DEFAULT_EXPIRE_TIME: 180,
```
Comment (raphj) [inline frontend/src/helper.js:92]: Should this be configurable? why 180?

File: frontend/src/container/Home.jsx (hunk 1 above)
Comment (gbonaspetti) [reply frontend/src/container/Home.jsx:264]: Personally I prefer it that way, but I was in doubt, maybe you're right, @celiamgtt what do you think?

Comment (celiamgtt) [reply frontend/src/container/Home.jsx:264]: Yes it would be better ;)

File: frontend/src/container/Home.jsx (hunk 2 above)
Comment (celiamgtt) [reply frontend/src/container/Home.jsx:267]: Not the text between brackets (imo)

File: frontend/src/container/Home.jsx (hunk 1 above)
Comment (gbonaspetti) [reply frontend/src/container/Home.jsx]: Changed.

File: frontend/src/container/Home.jsx (hunk 2 above)
Comment (gbonaspetti) [reply frontend/src/container/Home.jsx:267]: I agree with @celiamgtt

**Comment by inkhey** [comment]
Tested on branch.

**Pull request merged**